monitoring/v1/monitoring.proto

```diff
@@ -51,6 +51,71 @@ service MonitoringService {
             body: "*"
         };
     }
+
+    // Get the memory usage metrics for the deployment based on the given request.
+    // Required permissions:
+    // - monitoring.metrics.get on the deployment identified by the given deployment ID.
+    rpc GetDeploymentMemoryUsageMetrics(GetDeploymentMetricsRequest) returns (DeploymentMetrics) {
+        option (google.api.http) = {
+            get: "/api/monitoring/v1/metrics/memory"
+        };
+    }
+
+    // Get the CPU usage metrics for the deployment based on the given request.
+    // Required permissions:
+    // - monitoring.metrics.get on the deployment identified by the given deployment ID.
+    rpc GetDeploymentCPUUsageMetrics(GetDeploymentMetricsRequest) returns (DeploymentMetrics) {
+        option (google.api.http) = {
+            get: "/api/monitoring/v1/metrics/cpu"
+        };
+    }
+
+    // Get the disk usage metrics for the deployment based on the given request.
+    // Note: The response will contain an empty list for coordinators.
+    // Required permissions:
+    // - monitoring.metrics.get on the deployment identified by the given deployment ID.
+    rpc GetDeploymentDiskUsageMetrics(GetDeploymentMetricsRequest) returns (DeploymentMetrics) {
+        option (google.api.http) = {
+            get: "/api/monitoring/v1/metrics/disk"
+        };
+    }
+}
+
+// Request for getting deployment metrics
+message GetDeploymentMetricsRequest {
+    // ID of the deployment for which metrics are being requested.
+    // This is a required field.
+    string deployment_id = 1;
+    // Start time for the query.
+    // This is a required field.
+    google.protobuf.Timestamp start_at = 2;
+    // The end time for the query.
+    // This is a required field.
+    google.protobuf.Timestamp end_at = 3;
+    // Granularity defines the interval between each metric sample.
+    // This needs to be a valid duration.
+    // Example: 1m, 2m, 30s, 1d, etc.
+    // Defaults to 30s.
+    // This is an optional field.
+    string granularity = 4;
```

Comment:
This way the interface allow to 'get' many-many, imo we should restrict the output, just like in the proposal by Matt.

Reply:
IMO We can replace this 3 fields by a single string (enum) to prevent this

Reply:
Or start_at and end_at only and let us do an internal calculation which granularity we will return to prevent too much data?

Reply:
Yeah, we can restrict the time range (like return an error if the window is >1week)

Reply:
Updated